_posts/2023-10-06-code-coverage.md

```diff
@@ -0,0 +1,94 @@
+---
+title: "Code Coverage for Unit Tests"
+excerpt: "Code coverage helps in the easy maintenance of the codebase, exposure of bad code, and results in faster time to market."
+tags: QA Mobile Android iOS Tests
+authors:
+- Rafael Nascimento
+header:
+  teaser: /assets/images/post/code-coverage/code_coverage.png
+  teaser_alt: Code coverage
+category: QA
+---
+
+![](/assets/images/post/code-coverage/code_coverage.png)
+
+# Introduction
+
+Code coverage is the percentage of lines of your code that is being executed by your tests. This is an indicative of how safe your code is and how bug proof it can be while refactoring or adding new features. Code coverage helps in the easy maintenance of the codebase, exposure of bad code, and results in faster time to market. 
+
+This metric is measured using the following formula:
+
+**_Coverage Percentage = (number of unique lines of code executed / total number of lines of code) * 100._**
+
+# Coverage Criteria
+
+To measure the lines of code that are actually exercised by test runs, various criteria are taken into consideration. We have outlined below a few critical coverage criteria that companies use.
+
+1. **Function Coverage**: The functions in the source code that are called and executed at least once.
+
+2. **Statement Coverage**: The number of statements that have been successfully validated in the source code.
+
+3. **Path Coverage**: The flows containing a sequence of controls and conditions that have worked well at least once.
+
+4. **Branch or Decision Coverage**: The decision control structures (loops, for example) that have executed fine.
+
+5. **Condition Coverage**: The Boolean expressions that are validated and that executes both TRUE and FALSE as per the test runs.
+
+# What to test?
+
+When making your test strategy you should aim to set up your team for success by understanding what is important to test in your project and what is not. This should be discussed before undertaking any new feature work, and should be part of the DoD (definition of done) of a user story.
+
+You should keep in mind the following:
+
+* You should write unit test to all the business logic of features developed by your team, such as view model methods, use cases and utility classes should be tested. Etc.
```

Comment:
If it is only 1 bullet point, does it need to be as a list item? Perhaps merge it to the same sentence?

Reply:
Rephrased and removed bullet point